### PR TITLE
Human override operator takes control of the bot session

### LINKED ISSUE #229: [human override] Allow the user to pause/resume the bot
For a particular message, allow the user to pause the bot and take control of it. Forward a message uuid to the bot, so it can use it to append messages to the original query.

When the message is resolved, return control to the bot.

Allow the user to send multiple responses without resolving the message.

### PR DESCRIPTION
**Pull request by aeatencio**
Depends on instedd/aida#264
Resolves #229 
Resolves instedd/aida#262
Technical debt: #234

### CODE REVIEW

File: app/javascript/reducers/messages.js [hunk 1]
```diff
@@ -9,16 +9,46 @@ const initialState = {
...
+    items: omit(state.items, [messageId]),
+  }
+}
+
+const answerSuccess = (state, action) => {
+  const { messageId, answer } = action
+  const { items } = state
+
+  if (items) {
+    const message = { ...items[`${messageId}`] }
```
Comment (aeatencio) [inline app/javascript/reducers/messages.js]: This string interpolation doesn't feel right, but it was the only way to get rid of this error: 
image

File: app/javascript/actions/messages.js [hunk 2]
```diff
@@ -35,14 +36,34 @@ export const fetchMessages = () => (dispatch : T.Dispatch, getState : T.GetState
             .catch(error => dispatch(_messagesReceiveError()))
 }
 
-export const _answerMessage = (messageId : number, answer : string) : T.HumanOverrideMessageAction => ({
-  type: ANSWER,
+export const _answerMessageSuccess = (messageId : number, answer : string) : T.HumanOverrideMessageAction => ({
```
Comment (nekron) [inline app/javascript/actions/messages.js]: Maybe the problem with flow is due to having this messageId typed as integer. Wether is easy to change this type or not, or if it should be an integer or a string, is an entirely different question. Let's pair this tomorrow.

File: app/javascript/components/HumanOverrideChatWindow.jsx [hunk 3]
```diff
@@ -146,21 +163,41 @@ class InputMessage extends Component {
...
+  }
+
+  return <Paper
     zDepth={5}
     className={"chat-window"}>
       <ChatHeader title={message.data.name || 'Unknown'} subtitle={bot.name} />
-      <MessageList messages={[{id: message.id, text: message.data.message, sent: false, timestamp: message.created_at}]} />
-      <InputMessage onSend={sendMessage} ref={inputRef}/>
+      <MessageList messages={messages()} />
+      <InputMessage onSend={sendMessage} onResolve={resolveMessage} ref={inputRef}/>
```
Comment (nekron) [inline app/javascript/components/HumanOverrideChatWindow.jsx]: The resolve button should be on the ChatHeader according to the mockups, but we can move it later.

File: app/models/notification.rb [hunk 4]
```diff
@@ -1,4 +1,46 @@
...
+  end
+
+  def bot_forwarding_messages!(value)
+    raise 'bot_forwarding_messages only supported for a human_override notification' if notification_type != 'human_override'
+
+    data['bot_forwarding_messages'] = value
+  end
+
+  def bot_forwarding_messages?
+    data['bot_forwarding_messages'] || false
```
Comment (nekron) [inline app/models/notification.rb]: In ruby there is no need for the `|| false` clause as `nil` is a "falsey" value. `|| false` works because of that.

**Pull request merged**